libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationHelper.java

```diff
@@ -299,10 +300,9 @@ static NavigationIndices increaseIndex(RouteProgress routeProgress,
                                                                              List<StepIntersection> intersections) {
     List<Pair<StepIntersection, Double>> distancesToIntersections = new ArrayList<>();
     List<StepIntersection> stepIntersections = new ArrayList<>(intersections);
-    if (stepPoints.isEmpty()) {
-      return distancesToIntersections;
-    }
-    if (stepIntersections.isEmpty()) {
+    boolean invalidPointList = stepPoints.size() < TWO_POINTS;
+    boolean emptyIntersectionList = stepIntersections.isEmpty();
+    if (invalidPointList || emptyIntersectionList) {
       return distancesToIntersections;
```

Comment:
Could we return `Collections.emptyList()` to make it more explicit? If so we can move define `distancesToIntersections` down, close to where it's used. BTW we should check if there's any implication on returning an immutable list in this case 👀 https://stackoverflow.com/questions/5552258/collections-emptylist-vs-new-instance#answer-5552287